File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/Python/exercicio17.py
"""
17 - Faça uma função para criptografar uma frase digitada pelo usuário.
A criptografia deverá substituir todas as vogais por *. 
Utilize a função no programa principal. A frase poderá ter, no máximo, 50 caracteres.
"""

#FUNÇÃO CRIPTOGRAFIA
def criptografia(frase):
  for i in range(len(frase)):
    if frase[i] in 'AaEeIiOoUu':
      frase = frase.replace(frase[i], '*')

  print(frase)

#FUNÇÃO PRINCIPAL
frase = str(input('Digite uma frase: '))
criptografia(frase)

File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/TypeScript/Exercicio17.ts
/*
17 - Faça uma função para criptografar uma frase digitada pelo usuário.
A criptografia deverá substituir todas as vogais por *. 
Utilize a função no programa principal. A frase poderá ter, no máximo, 50 caracteres.
*/

function criptografia(frase: string) {
    frase = frase.replace(/a/gi, '*')
    frase = frase.replace(/e/gi, '*')
    frase = frase.replace(/i/gi, '*')
    frase = frase.replace(/o/gi, '*')
    frase = frase.replace(/u/gi, '*')
    console.log(frase)
}

//FUNÇÃO PRINCIPAL
const frase = 'Aprendendo JavaScript'
criptografia(frase)
